README.md

```diff
@@ -207,31 +162,125 @@ Users want to determine several types of the formatting width as following
+```
+
+### Constructor
+
+#### Syntax
+
+```javascript
+new Intl.DurationFormat(locales, options)
 ```
+
+#### Parameters
+
+* `locales` (`Array` | `String`): A locale string or a list of locale strings in decreasing order of preference.
+* `options` (optional `Object`): An object for configuring the behavior of the instance. It may have some or all of the following properties:
+  * `localeMatcher` (`String`): A string denoting which locale matching algorithm to use. Can be either `"best-fit"` or `"lookup"`. Defaults to `"best fit"`.
+  * `numberingSystem` (`String`): A string containing the name of the numbering system to be used for number formatting. Defaults to `undefined`.
+  * `style` (`String`): The base style to be used for formatting. This can be overriden per-unit by setting the more granular options. Can be one of `"long"`, `"short"` or `"narrow"`. Defaults to `"short"`.
+  * `years` (`String`): The style to be used for formatting years. Can be one of `"long"`, `"short"`, or `"narrow"`. Defaults to the value of `style`.
+  * `yearsDisplay` (`String`): When to display years. `"auto"` means that years won't be displayed if they are zero. Can be either `"always"` or `"auto"`. Defaults to `"auto"` if `years` is `undefined` and `"always"` otherwise.
+  * `months` (`String`): The style to be used for formatting months. Can be one of `"long"`, `"short"`, or `"narrow"`. Defaults to the value of `style`.
+  * `monthsDisplay` (`String`): When to display months. `"auto"` means that months won't be displayed if they are zero. Can be either `"always"` or `"auto"`. Defaults to `"auto"` if `months` is `undefined` and `"always"` otherwise.
+  * `weeks` (`String`): The style to be used for formatting weeks. Can be one of `"long"`, `"short"`, or `"narrow"`. Defaults to the value of `style`.
+  * `weeksDisplay` (`String`): When to display weeks. `"auto"` means that weeks won't be displayed if they are zero. Can be either `"always"` or `"auto"`. Defaults to `"auto"` if `weeks` is `undefined` and `"always"` otherwise.
+  * `days` (`String`): The style to be used for formatting days. Can be one of `"long"`, `"short"`, or `"narrow"`. Defaults to the value of `style`.
+  * `daysDisplay` (`String`): When to display days. `"auto"` means that days won't be displayed if they are zero. Can be either `"always"` or `"auto"`. Defaults to `"auto"` if `days` is `undefined` and `"always"` otherwise.
+  * `hours` (`String`): The style to be used for formatting hours. Can be one of `"long"`, `"short"`, or `"narrow"`. Defaults to the value of `style`.
+  * `hoursDisplay` (`String`): When to display hours. `"auto"` means that hours won't be displayed if they are zero. Can be either `"always"` or `"auto"`. Defaults to `"auto"` if `hours` is `undefined` and `"always"` otherwise.
+  * `minutes` (`String`): The style to be used for formatting minutes. Can be one of `"long"`, `"short"`, or `"narrow"`. Defaults to the value of `style`.
```

Comment:
There should be discussion here and in `seconds` (and in `hours` too if `'numeric'` is a valid option for `days`, which I think it should be) about how `'numeric'` is upgraded to `'2-digit'` if there's a `'numeric'` unit to its left.  So `{hours: 'numeric', minutes: 'numeric'}` really means `{hours: 'numeric', minutes: '2-digit'}`. At least that's how I think we agreed it would work. @sffc is that correct?